core/amber/src/main/python/core/models/marker.py

```diff
@@ -1,11 +1,70 @@
 from dataclasses import dataclass
+from pyarrow import Table
+from pandas import DataFrame
+from typing import Optional
+from .schema import Schema, AttributeType
+from .schema.attribute_type import FROM_PYOBJECT_MAPPING
 
 
 @dataclass
 class Marker:
     pass
 
 
+@dataclass
+class StartOfUpstream(Marker):
+    pass
+
+
 @dataclass
 class EndOfUpstream(Marker):
     pass
+
+
+@dataclass
+class State(Marker):
+    def __init__(
+        self, table: Optional[Table] = None, pass_to_all_downstream: bool = False
+    ):
+        if table is None:
+            self.data = {}
+            self.schema = Schema()
+            self.add("passToAllDownstream", pass_to_all_downstream)
+        else:
+            self.data = table.to_pandas().iloc[0].to_dict()
+            self.schema = Schema(table.schema)
+
+    def add(
+        self, key: str, value: any, value_type: Optional[AttributeType] = None
+    ) -> None:
+        self.data[key] = value
+        if value_type is not None:
+            self.schema.add(key, value_type)
+        else:
+            self.schema.add(key, FROM_PYOBJECT_MAPPING[type(value)])
+
+    def get(self, key: str) -> any:
+        return self.data[key]
+
+    def is_pass_to_all_downstream(self) -> bool:
+        return self.data["passToAllDownstream"]
+
```

Comment:
in the constructor, in the else branch, you are not initialziting `passToAllDownstream`. now you are getitem from a dict that may cause KeyError. try to avoid such access method and possible error.

Reply:
Fixed.